src/components/modals/VoteHistoryModal.vue

```diff
@@ -1,17 +1,43 @@
 <template>
   <Modal
     class="vote-history"
-    v-if="modals.voteHistory && session.voteHistory"
+    v-if="modals.voteHistory && (session.voteHistory || !session.isSpectator)"
     @close="toggleModal('voteHistory')"
   >
     <font-awesome-icon
       @click="clearVoteHistory"
       icon="trash-alt"
       class="clear"
       title="Clear history"
+      v-if="session.isSpectator"
     />
 
     <h3>Nomination history</h3>
+
+    <template v-if="!session.isSpectator">
+      <table>
+        <tr>
+          <td class="option">
+            <div class="option" @click="setRecordVoteHistory">
+              <font-awesome-icon
+                :icon="[
+                  'fas',
+                  session.isVoteHistoryAllowed ? 'check-square' : 'square'
+                ]"
+              />
+              Allow players to view this?
+            </div>
+          </td>
+          <td></td>
+          <td>
+            <div class="option" @click="clearVoteHistory">
+              <font-awesome-icon icon="trash-alt" />
+              Clear players vote histories
```

Comment:
```suggestion
              Clear for everyone
```
Talking about "players vote histories" here implies that the ST history is not cleared.